backend/IDE.BLL/Services/UserService.cs

```diff
@@ -162,5 +200,50 @@ private async Task<User> GetUserByIdInternal(int id)
                 .Include(u => u.Avatar)
                 .FirstOrDefaultAsync(u => u.Id == id);
         }
+
+        public async Task UpdateUserAvatar(ImageUploadBase64DTO imageUploadBase64DTO, int userId)
+        {
+            var imgSrc = await _imageUploader.UploadAsync(imageUploadBase64DTO.Base64);
+            var userEntity = await _context.Users.SingleOrDefaultAsync(u => u.Id == userId);
+
+            await _context.Images.AddAsync(new Image { Url = imgSrc });
+            await _context.SaveChangesAsync();
+
+            var imageId = await _context.Images.LastAsync();
```

Comment:
Don't use this, EF will automatically set the ID for the entity if you write above:
```
var img = new Image { Url = imgSrc };
await _context.Images.AddAsync(img);
await _context.SaveChangesAsync(); <- here EF will set Id

img.Id <- now you can get the Id of the created object
```